### DISCUSSION TITLE
Default branch name config somehow getting reset to `root`

### DISCUSSION THREAD

**Discussion by sivaraam** [git-for-windows/git#3578] [Q&A]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.34.1.windows.1
cpu: x86_64
built from commit: 2ca94ab318509b3c271e82889938816bad76dfea
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19043.1348]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: Notepad++
Custom Editor Path:
Default Branch Option: main
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: LFOnly
Bash Terminal Option: MinTTY
Git Pull Behavior Option: FFOnly
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Enabled
Enable Pseudo Console Support: Enabled
Enable FSMonitor: Enabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Not really sure. I use [Windows Terminal](https://github.com/Microsoft/Terminal) if that's in anyway useful.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Windows Terminal with [PowerShell Core](https://github.com/powershell/powershell) v6.2.2 (side note: I really need to upgrade PowerShell Core, it seems).

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
mkdir throwaway
cd throwaway
git init
git status
```
 - What did you expect to occur after running these commands?

I was expecting the branch name to be `main` which is what I configured as the default branch name during an upgrade.

 - What actually happened instead?

I get `root` as the default branch name.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

This doesn't seem specific to any repository.

### More info
The issue isn't weird in one aspect. The `root` branch name doesn't come out of no where. It does seem to be coming through the user config:

```
> git config --show-origin --get-all init.defaultbranch
file:C:/Program Files/Git/etc/gitconfig main
file:C:/Users/<USER_NAME>/.gitconfig root
```

The weird part is, I could swear I never configured the default branch to `root` myself. Heck, I don't even like that name for a branch let alone set that as the default. It's really confusing how this magical config gets set from time to time. Yeah, you read it right. I've observed this before. Corrected it (by removing the user specific config I believe) but it came back again. Is there some sort of way to identify how this config is getting set? That would really helpful in identifying the issue.

**Comment by dscho** [comment]
You could use [Trace2](https://git-scm.com/docs/api-trace2) to track all invocations, to see whether there is any suspicious recurrent `git config --global init.defaultBranch root` call, then deduce from the context whether it was an automated thing, or part of, say, a `make` invocation.
